pymc3/distributions/distribution.py

```diff
@@ -115,13 +128,187 @@ def logcdf(op, var, rvs_to_values, *dist_params, **kwargs):
+        )
+
+    # The numeric/symbolic resize tuple can be created using model.RV_dim_lengths
+    resize_shape = tuple(model.dim_lengths[dname] for dname in dims[:ndim_resize])
+    return ndim_resize, resize_shape, dims
+
+
+def _resize_from_observed(
+    observed, ndim_implied: int
+) -> Tuple[int, StrongSize, Union[np.ndarray, Variable]]:
+    """Determines a potential resize shape from observations.
+
+    Parameters
+    ----------
+    observed : scalar, array-like
+        The value of the `observed` kwarg to the RV creation.
+    ndim_implied : int
+        Number of RV dimensions that were implied from its inputs alone.
+
+    Returns
+    -------
+    ndim_resize : int
+        Number of dimensions that should be added through resizing.
+    resize_shape : array-like
+        The shape of the new dimensions.
+    observed : scalar, array-like
+        Observations as numpy array or `Variable`.
+    """
+    if not hasattr(observed, "shape"):
+        observed = pandas_to_array(observed)
+    ndim_resize = observed.ndim - ndim_implied
+    resize_shape = tuple(observed.shape[d] for d in range(ndim_resize))
+    return ndim_resize, resize_shape, observed
+
+
 class Distribution(metaclass=DistributionMeta):
     """Statistical distribution"""
 
     rv_class = None
     rv_op = None
 
-    def __new__(cls, name, *args, **kwargs):
+    def __new__(
+        cls,
+        name: str,
+        *args,
+        rng=None,
+        dims: Optional[Dims] = None,
+        testval=None,
+        observed=None,
+        total_size=None,
+        transform=UNSET,
+        **kwargs,
+    ) -> RandomVariable:
+        """Adds a RandomVariable corresponding to a PyMC3 distribution to the current model.
+
+        Note that all remaining kwargs must be compatible with ``.dist()``
+
+        Parameters
+        ----------
+        cls : type
+            A PyMC3 distribution.
+        name : str
+            Name for the new model variable.
+        rng : optional
+            Random number generator to use with the RandomVariable.
+        dims : tuple, optional
+            A tuple of dimension names known to the model.
+        testval : optional
+            Test value to be attached to the output RV.
+            Must match its shape exactly.
+        observed : optional
+            Observed data to be passed when registering the random variable in the model.
+            See ``Model.register_rv``.
+        total_size : float, optional
+            See ``Model.register_rv``.
+        transform : optional
+            See ``Model.register_rv``.
+        **kwargs
+            Keyword arguments that will be forwarded to ``.dist()``.
+            Most prominently: ``shape`` and ``size``
+
+        Returns
+        -------
+        rv : RandomVariable
+            The created RV, registered in the Model.
+        """
+
         try:
             from pymc3.model import Model
```

Comment:
My guess would be to prevent a circular import.
If it works that'd be great, because we could use `Model` in some type hints.

But it's out of scope for this PR.